packages/webpack-cli/lib/webpack-cli.js

```diff
@@ -2324,6 +2324,27 @@ class WebpackCLI {
       return;
     }
 
+    const signals = ["SIGINT", "SIGTERM"];
+    let needForceShutdown = false;
+
+    signals.forEach((signal) => {
+      const listener = () => {
+        if (needForceShutdown) {
+          process.exit();
+        }
+
+        this.logger.info("Gracefully shutting down. To force exit, press ^C again. Please wait...");
+
+        needForceShutdown = true;
+
+        compiler.close(() => {
+          process.exit();
+        });
+      };
+
+      process.on(signal, listener);
+    });
```

Comment:
We should run it only when `cache` is not `false`/`undefined`

Reply:
Done. Will do for watch as well then we are good to go.